addon/components/connect.js

```diff
@@ -37,9 +38,16 @@ export default (stateToComputed, dispatchToActions=() => ({})) => {
           });
         }
 
-        this.actions = Object.assign({},
-          this.actions, dispatchToActions.call(this, redux.dispatch.bind(redux))
-        );
+        if (typeof dispatchToActions === 'function') {
+          this.actions = Object.assign({},
+            this.actions, dispatchToActions.call(this, redux.dispatch.bind(redux))
+          );
+        }
+
+        if (typeof dispatchToActions === 'object') {
```

Comment:
I just want to note that `typeof [] === 'object'` evaluates to true, so you might actually be better off going with `Ember.typeOf(dispatchToActions) === 'object`. If you want to also support Ember.Object's then you could do something like `['instance', 'object'].indexOf(typeOf(dispatchToActions)) !== -1`.